Improve the doc of bitcoin staking script (#618)
babylonchain/babylon: docs/staking-script.md

7. (WIP) restakable, meaning that the same bitcoin can be staked to secure
   multiple PoS chains and earn multiple PoS yields

In the entire Bitcoin staking process, two parties are involved: one is called 
the Bitcoin Staker, and the other is called the Finality Provider.

> - **Bitcoin Staker**: A Bitcoin Staker is a person who owns Bitcoin and chooses 
> to lock up their Bitcoin for a set period of time to earn yields.
> - **Finality Provider**: A Finality Provider is the service that provides security 
> assurance to the proof-of-stake chain.

Reviewer: ```suggestion
- **Bitcoin Staker**: A Bitcoin Staker is a person who owns Bitcoin and chooses 
to lock up their Bitcoin for a set period of time to secure a PoS chain.
- **Finality Provider**: A Finality Provider is an entity that votes in the finality round
to provide security assurance to the PoS chain.
```

> The Bitcoin Staker can choose a specific Finality Provider with 
> which to lock (stake) their Bitcoins."

Reviewer: ```suggestion
The Bitcoin Staker can choose a specific Finality Provider to which 
to they can delegate (stake) their locked Bitcoin."
```

> to they can delegate (stake) their locked Bitcoin.

Reviewer: ```suggestion
they can delegate the voting power stemming from their locked Bitcoin.
```

Reviewer: Don't want this to be confused to delegating custody of the locked BTC.

Author: Rewrite the sentence on the last commit
